azurerm/internal/services/containers/tests/kubernetes_cluster_scaling_resource_test.go

```diff
@@ -238,6 +238,7 @@ func testAccAzureRMKubernetesCluster_autoScalingNodeCountUnset(t *testing.T) {
 					resource.TestCheckResourceAttr(data.ResourceName, "default_node_pool.0.max_count", "4"),
 					resource.TestCheckResourceAttr(data.ResourceName, "default_node_pool.0.enable_auto_scaling", "true"),
 					resource.TestCheckResourceAttr(data.ResourceName, "auto_scaler_profile.0.max_graceful_termination_sec", "600"),
+					resource.TestCheckResourceAttr(data.ResourceName, "auto_scaler_profile.0.new_pod_scale_up_delay", "10s"),
```

Comment:
it appears this defaults to `0s` in the API:

```
------- Stdout: -------
=== RUN   TestAccAzureRMKubernetesCluster_autoScalingNodeCountUnset
=== PAUSE TestAccAzureRMKubernetesCluster_autoScalingNodeCountUnset
=== CONT  TestAccAzureRMKubernetesCluster_autoScalingNodeCountUnset
testing.go:684: Step 0 error: Check failed: Check 6/13 error: azurerm_kubernetes_cluster.test: Attribute 'auto_scaler_profile.0.new_pod_scale_up_delay' expected "10s", got "0s"
--- FAIL: TestAccAzureRMKubernetesCluster_autoScalingNodeCountUnset (584.16s)
FAIL
```

As such we may need to explicitly define the default value, rather than leaning on the Azure API here

Reply:
The default has always been defined as 0. I followed up to update the documentation.

So, the TF provider should set it as 0 as well if possible.

Reply:
in which case we'll need to update this assertion line in the test to account for that, since this is assuming the default value is `10s` when it's `0s`

```suggestion
					resource.TestCheckResourceAttr(data.ResourceName, "auto_scaler_profile.0.new_pod_scale_up_delay", "0s"),
```